### PR TITLE
ci-operator/*: Refactor CRI-O CI and fix the Python issue

### PR DESCRIPTION
**Pull request by sohankunkerkar**
CRI-O CI has been broken for the last couple of days. This PR fixes the current Python-related issue and refactors CRI-O CI to handle its own GCP infra bits, avoiding much reliance on the UPI installer setup.

### CODE REVIEW

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by rphillips** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

**Comment by sohankunkerkar** [comment]
/pj-rehearse pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml [hunk 1]
```diff
@@ -25,6 +25,18 @@ images:
     COPY . .
   from: base
   to: crio-crio-base-src
+- dockerfile_literal: |
+    FROM base
+    RUN dnf install -y jq git openssl python3.11 google-cloud-cli-447.0.0-1 gzip jq unzip openssh-clients openssl bind-utils util-linux xz && \
+        alternatives --set python3 /usr/bin/python3.11 && \
```
Comment (sohankunkerkar) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: I doubt this will work. I got an error in the previous CI run.

**Comment by sohankunkerkar** [comment]
@jianlinliu  @jianli-wei could you PTAL?

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml [hunk 2]
```diff
@@ -6,26 +11,41 @@ build_root:
...
       && rm -f /tmp/google-cloud-sdk.tar.gz
     ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
     WORKDIR /src
     COPY . .
-  from: src
+  from: base
   to: crio-crio-base-src
+- dockerfile_literal: |
+    FROM base
+    RUN dnf install -y jq git openssl python3.11 google-cloud-cli-447.0.0-1 gzip jq unzip openssh-clients openssl bind-utils util-linux xz && \
```
Comment (saschagrunert) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:30]: Follow-up: Can we use the latest python 3 and gcloud cli here? I fear that those versions may get outdated in the future and we don’t see it.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml [hunk 3]
```diff
@@ -6,26 +11,41 @@ build_root:
...
+        ln -sf /usr/bin/python3.11 /usr/bin/python3 && \
```
Comment (saschagrunert) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: For what do we need this symlink?

Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: Python 3.9 is currently the default, so we need to ensure that the `google-cloud SDK` picks the latest version of Python.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml (hunk 2 above)
Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:30]: Yeah, that's a good point.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml (hunk 3 above)
Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: https://github.com/google-cloud-sdk-unofficial/google-cloud-sdk/blob/master/install.sh#L133

Comment (saschagrunert) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: Sounds good!

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml (hunk 2 above)
Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:30]: Let me get the CI working, and then I'll update all dependencies to the latest version.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml (hunk 3 above)
Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: I tried that approach but it didn't work: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/45338/rehearse-45338-pull-ci-cri-o-cri-o-main-ci-cgroupv2-integration/1722391694151782400

Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: I think I can use this to fix that issue:
```sh
RUN alternatives --install /usr/bin/python3 python3 /usr/bin/python3.11 1
RUN alternatives --set python3 /usr/bin/python3.11
```

Comment (sohankunkerkar) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: can I take this as a follow-up?

Comment (saschagrunert) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-main__ci.yaml:31]: Sure.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**